src/StackExchange.Redis/Configuration/DefaultOptionsProvider.cs

```diff
@@ -49,10 +49,12 @@ public static void AddProvider(DefaultOptionsProvider provider)
         /// </summary>
         public virtual bool IsMatch(EndPoint endpoint) => false;
 
+        private static readonly DefaultOptionsProvider s_DefaultProvider = new DefaultOptionsProvider();
```

Comment:
I'm a little uncomfortable here because there's a chance this becomes mutable at some point down the road (and we're not returning a new one each time). Thoughts?

Reply:
@NickCraver the `AddProvider` API already kinda assumes that it isn't, since that is instance based; so if it comes up as a mutability issue: we already have a bigger problem

Reply:
@NickCraver  for example, *today* you could just do `DefaultOptionsProvider.AddProvider(new DefaultOptionsProvider());` - and that needs to work

Reply:
I agree that'd happen _if you did that_ due to matching, but today every caller would be getting their own instance, no? I'm not sure how shifting this changes the example, but does return the same instance for all.

I'm saying this because `clientName` is stateful, and has the potential to change on subsequent caller/connections as the environment does (e.g. restarting your Azure app after an environment change isn't uncommon).

Reply:
Then any subclass already has this problem?

Reply:
hmm, yeah it could indeed, good point - this just makes it worse for the default Azure case.

Reply:
I'm happy to revert the default instance for now and take that point separately; this isn't a hill for today

Reply:
If we can please - I know this will break some scenarios (silently reporting old IDs, few would notice)

Reply:
@NickCraver done